src/libxively/mqtt/logic/xi_mqtt_logic_layer.c

```diff
@@ -605,6 +605,11 @@ xi_mqtt_logic_layer_close_externally( void* context, void* data, xi_state_t in_o
     /* let's stop the current task */
     if ( layer_data->current_q0_task != 0 )
     {
+        if ( NULL != layer_data->current_q0_task->timeout.ptr_to_position )
```

Comment:
This is the actual bugfix. The rationale behind:
This part of code is the close_externally, which is the destruction of the layerchain. Ongoing task is executed at destruction, to notify the application callbacks about task progress (PUBLISH / SUBSCRIBE failure). But in this case the task is a timed task which sits inside the event dispatcher's timed event's queue.
The assert in question is checking the idea behind: keepalive task should be executed by event scheduler which actually resets the timed event. But here not the event scheduler executes the task but the mqtt logic layer's destructor function. So here we cancel the timed event before execution which will silence the assert and the idea / contract guarded by the asserts kept valid.